frontend/packages/dev-console/src/components/topology/redux/reducer.ts

```diff
@@ -1,19 +1,30 @@
 import { Map } from 'immutable';
 import { merge } from 'lodash';
 import { TopologyAction, Actions } from './action';
-import { TOPOLOGGY_FILTERS_LOCAL_STORAGE_KEY, DEFAULT_TOPOLOGY_FILTERS } from './const';
+import {
+  TOPOLOGY_DISPLAY_FILTERS_LOCAL_STORAGE_KEY,
+  DEFAULT_TOPOLOGY_FILTERS,
+  TOPOLOGY_SEARCH_FILTER_KEY,
+} from './const';
 
 export type State = Map<string, any>;
 
 export const getDefaultTopologyFilters = () => {
-  const filters = localStorage.getItem(TOPOLOGGY_FILTERS_LOCAL_STORAGE_KEY);
-  if (filters) {
-    return merge({}, DEFAULT_TOPOLOGY_FILTERS, JSON.parse(filters));
+  const displayFilters = localStorage.getItem(TOPOLOGY_DISPLAY_FILTERS_LOCAL_STORAGE_KEY);
+  const searchQuery = new URLSearchParams(window.location.search).get(TOPOLOGY_SEARCH_FILTER_KEY);
+
+  if (displayFilters && searchQuery) {
```

Comment:
What if we have one or the other but not both? We still should get these values.

Reply:
Okay, just fixed it. Earlier I had the implementation with `displayFilters || searchQuery` which resulted in error if `displayFilter` was not available and there is `searchQuery`. So fixed that one but didn't realize that the `displayFilter` will be reset with default in `LocalStorage` every time when `searchQuery` is not there.

Reply:
I've tested it for both scenarios now.